src/features/debugger/debugMessage.ts

```diff
@@ -95,15 +98,21 @@ export async function debugMessage(
   if (ismCheckResult.status && ismCheckResult.description) return { ...ismCheckResult, ...details };
   else details.ismDetails = ismCheckResult.ismDetails;
 
-  const gasCheckResult = await tryCheckIgpGasFunded(
-    msgId,
-    originProvider,
-    deliveryResult.gasEstimate,
-    totalGasAmount,
-  );
-  if (gasCheckResult?.status && gasCheckResult?.description)
-    return { ...gasCheckResult, ...details };
-  else details.gasDetails = gasCheckResult?.gasDetails;
+  // TODO support for non-default IGP gas checks here
+  // Disabling for now for https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3668
+  // Also skipping if the message is still very new otherwise this raises premature
+  // underfunded errors when in fact payment was made
```

Comment:
honestly I think this whole approach is too fine-grained
instead we can use the `quoteDispatch` function on `HyperlaneCore` to calculate what the fee should be

Reply:
then we dont need to track IGP addresses here/on warpcore/in the registry

Reply:
Agreed, our SDK tooling has improved since this was first written. Added a note about this here: https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/1999